modules/tinymce/src/plugins/accordion/main/ts/core/Actions.ts

```diff
@@ -29,4 +27,55 @@ const insertAccordion = (editor: Editor): void => {
   }
 };
 
-export { insertAccordion };
+const toggleDetailsElement = (details: HTMLDetailsElement, state?: boolean): boolean => {
+  const shouldOpen = state ?? !Utils.isOpen(details);
+  if (shouldOpen) {
+    details.setAttribute('open', 'open');
+    details.setAttribute('data-mce-open', 'open');
+  } else {
+    details.removeAttribute('open');
+    details.removeAttribute('data-mce-open');
+  }
+  return shouldOpen;
+};
+
+const toggleAccordion = (editor: Editor, state?: boolean): void => {
+  const details = Utils.getSelectedDetails(editor);
+  if (details) {
+    Events.fireToggleAccordionEvent(editor, details, toggleDetailsElement(details, state));
+  }
+};
+
+const removeAccordion = (editor: Editor): void => {
+  const details = Utils.getSelectedDetails(editor);
+  if (!details) {
+    return;
+  }
+
+  const { nextSibling } = details;
+  if (nextSibling) {
+    editor.selection.select(nextSibling, true);
+    editor.selection.collapse(true);
+  } else {
+    Utils.insertAndSelectParagraphAfter(editor, details);
+  }
+
+  details.remove();
+};
+
+const toggleAllAccordions = (editor: Editor, state?: boolean): void => {
+  const accordions = Array.from(editor.getBody().querySelectorAll('details'));
+  if (accordions.length === 0) {
+    return;
+  }
+  const shouldOpen = state ?? !Utils.isOpen(accordions[0]);
```

Comment:
Shouldn't each accordion invert the current state when no state is applied instead of depending on the state of the first one?

Reply:
Makes sense. Triggering the desired state is still possible with the additional argument, so by default it can toggle each accordion's state separately.